src/duckdb.go

```diff
@@ -45,15 +48,20 @@ func NewDuckdb(config *Config) *Duckdb {
 
 	switch config.StorageType {
 	case STORAGE_TYPE_S3:
-		query := "CREATE SECRET aws_s3_secret (TYPE S3, KEY_ID '$accessKeyId', SECRET '$secretAccessKey', REGION '$region', ENDPOINT '$endpoint', SCOPE '$s3Bucket')"
-		_, err = duckdb.ExecContext(ctx, query, map[string]string{
-			"accessKeyId":     config.Aws.AccessKeyId,
-			"secretAccessKey": config.Aws.SecretAccessKey,
-			"region":          config.Aws.Region,
-			"endpoint":        config.Aws.S3Endpoint,
-			"s3Bucket":        "s3://" + config.Aws.S3Bucket,
-		})
-		PanicIfError(err)
+		duckdb.setAwsCredentials(ctx)
+		ticker := time.NewTicker(10 * time.Minute)
+		time.Tick(10 * time.Minute)
+		go func() {
+			for {
+				select {
+				case <-ticker.C:
+					duckdb.setAwsCredentials(ctx)
+				case <-duckdb.refreshQuit:
+					ticker.Stop()
+					return
+				}
+			}
+		}()
```

Comment:
This refresh logic is required because duckdb doesn't refresh credentials. Ref - https://github.com/duckdb/duckdb-aws/issues/26


TODO: refactor to provide refresh interval as a config.